sycl/test/tools/Inputs/mock_renderd_access.c

```diff
@@ -1,36 +1,57 @@
 #define _GNU_SOURCE
 
 #include <errno.h>
+#include <glob.h>
 #include <stdarg.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/stat.h>
 
-int stat(const char *pathname, struct stat *statbuf) {
-  const char *mock_mode = getenv("MOCK_STAT_MODE");
-  if (strstr(pathname, "renderD128")) {
-    if (mock_mode && strcmp(mock_mode, "notfound") == 0) {
-      errno = ENOENT;
-      return -1;
-    }
-    if (mock_mode && strcmp(mock_mode, "exists") == 0) {
-      memset(statbuf, 0, sizeof(*statbuf));
-      statbuf->st_mode = S_IFCHR | 0666;
-      return 0;
+int glob(const char *pattern, int flags, int (*errfunc)(const char *, int),
+         glob_t *pglob) {
+  const char *mock_mode = getenv("MOCK_GLOB_MODE");
+  if (mock_mode && strcmp(mock_mode, "exists") == 0) {
```

Comment:
Maybe I'm nitpicking because it's afternoon and I'm getting bored with my refactoring work but, should we use `strncmp` and `strndup` instead of `strcmp` and `strdup`? `strncmp` would be safer since we are reading a character array from an environment variable...

Reply:
I think in this case it is safe to use strcmp and strdup, because getenv is guaranteed to return null-terminated string and I compare it only with string literals (also null-terminated), so we are dealing with null-terminated strings only and there is no danger of buffer overrun.